sqlmesh/core/scheduler.py

```diff
@@ -212,25 +214,38 @@ def evaluate(
             batch_index=batch_index,
             **kwargs,
         )
-        try:
-            self.snapshot_evaluator.audit(
-                snapshot=snapshot,
-                start=start,
-                end=end,
-                execution_time=execution_time,
-                snapshots=snapshots,
-                deployability_index=deployability_index,
-                wap_id=wap_id,
-                **kwargs,
+        audit_results = self.snapshot_evaluator.audit(
+            snapshot=snapshot,
+            start=start,
+            end=end,
+            execution_time=execution_time,
+            raise_exception=False,
+            snapshots=snapshots,
+            deployability_index=deployability_index,
+            wap_id=wap_id,
+            **kwargs,
+        )
+
+        audit_error_to_raise: t.Optional[AuditError] = None
+        for audit_result in [result for result in audit_results if result.count]:
```

Comment:
Let's make this a generator?

```suggestion
        for audit_result in (result for result in audit_results if result.count):
```